src/imperative/cached_op.cc

```diff
@@ -784,7 +784,7 @@ OpStatePtr CachedOp::DynamicForward(
   states.reserve(idx.num_nodes());
   std::vector<NDArray*> arrays;
   arrays.reserve(buff.size());
-  for (size_t i = 0; i < buff.size(); ++i) arrays.push_back(&buff[i]);
+  for (auto& i : buff) arrays.push_back(&i);
```

Comment:
I'd change it to `const auto& item : buff` to avoid confusing `i` with an index var (like in the loop below)

Reply:
Good call, updated with some more specific naming.